app/controllers/catalog_controller.rb

```diff
@@ -9,5 +9,16 @@ class CatalogController < ApplicationController
       :rows => 20 
     }
   end
+  
+  def index
+    super
+    Rails.cache.read_multi(@document_list.map(&:ci_id)).tap do |ci_ids_to_proxy_urls|
+      @ids_to_cached_proxy = Hash[
+        @document_list.map do |document|
+          [document.id, ci_ids_to_proxy_urls[[document.ci_id]]]
+        end
+      ]
+    end
+  end
```

Comment:
I think there may be a better home for this logic in either `SolrDocument#proxy_src` or in `MediaController#show`.